src/main/java/com/kitaab/hisaab/ledger/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,212 @@
+                        ExceptionEnum.UNABLE_TO_SAVE_DATA
+                ));
+
+        log.debug("Saved token details for user: {} with document id : {}", username, metadata.getId());
+
+        emailService.sendEmail(emailDetails, EmailTypeEnum.RESET_PASSWORD)
+                .whenComplete((re, exc) -> {
+                    if (exc == null) {
+                        log.info("Email is sent to user");
+                    }
+                    throw new FlowBreakerException(ExceptionEnum.UNABLE_TO_SEND_EMAIL.getMessage(),
+                            ExceptionEnum.UNABLE_TO_SEND_EMAIL);
+                });
+        return new SuccessResponse(HttpStatus.CREATED, ApplicationConstants.PASSWORD_RESET_MAIL_SENT);
+    }
+
+    @Override
+    public SuccessResponse resetPassword(String authToken, String newPassword) {
+        String username = jwtService.extractUsername(authToken);
+        log.info("Trying to change password for user: {}", username);
+        var tokenMetadata = Optional.ofNullable(tokenMetadataRepository.findByUsernameOrderByCreatedAtDesc(username))
+                .orElseThrow(() -> new FlowBreakerException(ExceptionEnum.USERNAME_NOT_FOUND.getFormattedMessage(username),
+                        ExceptionEnum.UNABLE_TO_SAVE_DATA));
+
+        log.debug("Validating password reset token");
+        Boolean isValid = jwtService.validateToken(tokenMetadata.getToken(), username);
+        if (isValid && Objects.equals(tokenMetadata.getToken(), authToken)) {
+            log.debug("Password reset token is valid for user:{}", username);
+            User user = Optional.ofNullable(userRepository.findByUsername(username)).orElseThrow(
+                    () -> new FlowBreakerException(ExceptionEnum.USERNAME_NOT_FOUND.getFormattedMessage(username),
+                            ExceptionEnum.USERNAME_NOT_FOUND)
+            );
+            user.setPassword(encoder.encode(newPassword));
+            user = Optional.of(userRepository.save(user)).orElseThrow(
+                    () -> new FlowBreakerException(ExceptionEnum.UNABLE_TO_SAVE_DATA.getFormattedMessage(),
+                            ExceptionEnum.UNABLE_TO_SAVE_DATA));
+            log.debug("Password is reset for the user: {}", user.getUsername());
+        } else {
+            log.error(ExceptionEnum.PASSWORD_RESET_TOKEN_INVALID.getMessage());
+            throw new FlowBreakerException(ExceptionEnum.PASSWORD_RESET_TOKEN_INVALID.getMessage(),
+                    ExceptionEnum.PASSWORD_RESET_TOKEN_INVALID);
+        }
+        return new SuccessResponse(HttpStatus.OK,
+                MessageFormat.format(ApplicationConstants.PASSWORD_RESET_SUCCESS, username));
+    }
+
+    @Override
+    public SuccessResponse getUsers(String usernamePrefix) {
+        log.info("Finding all Users starting with Username : {}", usernamePrefix);
+        List<User> users = userRepository.findByUsernameStartsWith(usernamePrefix);
```

Comment:
List of users should not show the currently logged in user.

Reply:
updated code for same